enzyme/tools/enzyme-tblgen/enzyme-tblgen.cpp

```diff
@@ -1225,55 +1228,82 @@ size_t fwd_call_args(TGPattern &pattern, size_t actArg,
     // and based on that get the fp/int + scalar/vector type
-    const auto typeOfArg = typeMap.lookup(pos);
-    if (typeOfArg == argType::len) {
-      const auto out = (Twine("len_") + name).str();
-      result.append((Twine("len_") + name).str());
-    } else if (typeOfArg == argType::fp) {
+    const auto ty = typeMap.lookup(pos);
+    if (ty == len) {
+      result.append((Twine("arg_") + name).str());
+    } else if (ty == fp) {
       if (pos == actArg) {
         result.append((Twine("d_") + name).str());
       } else {
-        result.append((Twine("fp_") + name).str());
+        result.append((Twine("arg_") + name).str());
+        // result.append((Twine("fp_") + name).str());
       }
-    } else if (typeOfArg == argType::vincData) {
+    } else if (ty == vincData) {
       auto nextName = nameVec[pos + 1];
       // get the position of the argument in the primary blas call
       auto nextArgPosition = nameMap.lookup(nextName);
       // and based on that get the fp/int + scalar/vector type
       auto typeOfNextArg = typeMap.lookup(nextArgPosition);
-      assert(typeOfNextArg == argType::vincInc);
+      assert(typeOfNextArg == vincInc);
       if (pos == actArg) {
         result.append((Twine("d_") + name + ", true_" + nextName).str());
       } else {
-        result.append((Twine("data_") + name + ", " + nextName).str());
+        result.append((Twine("arg_") + name + ", arg_" + nextName).str());
       }
       pos++; // extra ++ due to also handling vincInc
-    } else if (typeOfArg == argType::vincInc) {
+    } else if (ty == vincInc) {
       // might come without vincData, e.g. after DiffeRet
       result.append(name);
+    } else if (ty == mldData) {
+      auto nextName = nameVec[pos + 1];
+      // get the position of the argument in the primary blas call
+      auto nextArgPosition = nameMap.lookup(nextName);
+      // and based on that get the fp/int + scalar/vector type
+      auto nextTy = typeMap.lookup(nextArgPosition);
+      assert(nextTy == mldLD);
+      if (pos == actArg) {
+        result.append((Twine("d_") + name + ", true_" + nextName).str());
+      } else {
+        result.append((Twine("arg_") + name + ", arg_" + nextName).str());
+      }
+      pos++; // extra ++ due to also handling mldLD
+    } else if (ty == mldLD) {
+        // might come without mldData, e.g. after DiffeRet
+        // coppied from vincInc, but should verify if actually needed
+        result.append((Twine("arg_") + name).str());
+    } else if (ty == cblas_layout) {
+      // TODO: based on byRef
+    } else if (ty == trans || ty == diag || ty == uplo || ty == side){
+        result.append((Twine("arg_") + name).str());
     } else {
-      llvm::errs() << "name: " << name << " typename: " << typeOfArg << "\n";
-      llvm_unreachable("unimplemented input type!");
+      // TODO: impl
+      // llvm::errs() << "name: " << name << " typename: " << ty << "\n";
+      // llvm_unreachable("unimplemented input type!");
```

Comment:
should this be removed?

Reply:
changed from comment to code, better be safe.